_posts/2022-12-23-aws-lambda-snapstart-spring-cloud-function.md

```diff
@@ -0,0 +1,123 @@
+You can enable it, but only for published Lambda versions.
+This means that it only works for versions that are published on the AWS account, and that it is not implemented on the $[LATEST] tag. If you want to make use of Lambda SnapStart, be sure to do so on a published version.
+The snapshot of your Lambda is created upon the version publishing process.
+
+{:refdef: style="text-align: center;"}
+<img src="{{ '/img/2022-12-23-aws-lambda-snapstart-spring-cloud-function/snapstart-overview.png' | prepend: site.baseurl }}" alt="SnapStart overview" class="image center" style="margin:0px auto; max-width:100%">
+_SnapStart overview - snapshot gets created during version publishing - [AWS Lambda SnapStart (AWS re:Invent 2022)](https://www.youtube.com/watch?v=ZbnAithBNYY){:target="_blank" rel="noopener noreferrer"}_
+{: refdef}
+
+### Pricing
+The SnapStart feature comes with AWS Lambda and has no additional pricing.
+
+### Limitations
+While SnapStart is a great feature and can save time in Lambda cold starts, it also comes with its limitations.
+SnapStart currently does not support the following features and services:
+- [provisioned concurrency](https://docs.aws.amazon.com/lambda/latest/dg/provisioned-concurrency.html){:target="_blank" rel="noopener noreferrer"}
+- arm64 architecture
+- the [Lambda Extensions API](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-extensions-api.html){:target="_blank" rel="noopener noreferrer"}
+- [EFS](https://aws.amazon.com/efs/){:target="_blank" rel="noopener noreferrer"}
+- [X-Ray](https://aws.amazon.com/xray/){:target="_blank" rel="noopener noreferrer"}
+- Ephemeral storage up to 512 MB
+- Limited to Java 11 runtime
+
+#### Uniqueness
+SnapStart always requires your snapshot to be unique. 
+This means that if you have initialization code which generates unique content, it might not always be unique in the snapshot once it is restored in other Lambda invocations.
+The goal is to generate this content after the initialization process, so it is not part of the snapshot.
+Luckily, AWS has provided a [documentation page](https://docs.aws.amazon.com/lambda/latest/dg/snapstart-uniqueness.html){:target="_blank" rel="noopener noreferrer"} in which they provide best practices on how to tackle that problem.
+They even came up with a [SpotBugs plugin](https://github.com/aws/aws-lambda-snapstart-java-rules){:target="_blank" rel="noopener noreferrer"}  which finds potential issues in your code that could prevent SnapStart from working correctly.
+
+#### Networking
+Network connections are not being shared across different environments.
+Thus, if network connections (for example, to other AWS services such as an RDS or SQS) are instantiated in the initialization phase, they will not be shared and will most likely fail when the snapshot is being used later again.
+Although most popular frameworks have automatic database connection retries, it is worth the time to make sure that it works correctly.
+
+#### Ephemeral data
```

Comment:
How does this affect your application? This will also make the init phase "do less" from an application standpoint. It sounds like the SnapStart fixes the "loading time" of e.g. Spring but doesn't really do much for the actual application context?